packages/cli/rules/bucket.cursor.md

```diff
@@ -0,0 +1,365 @@
+- Implement proper error fallbacks when feature flag services are unavailable
+- Use environment-specific configuration for development, staging, and production
+- Leverage Bucket CLI for generating type-safe feature definitions
+
+Feature Targeting and Segmentation
+
+- Implement proper user and company context data for targeting
+- Design effective targeting rules based on user attributes
+- Create and manage reusable segments for consistent targeting
+- Use hierarchical targeting rules for complex user populations
+- Monitor and analyze targeting effectiveness
+- Follow best practices for exclusion and inclusion patterns
+
+Remote Configuration
+
+- Define type-safe configuration payloads for feature flags
+- Access and use configuration values correctly in components
+- Design fallback values for configuration when services are unavailable
+- Implement progressive enhancement patterns with remote configuration
+- Test with various configuration scenarios before deployment
+- Use configuration for both functional and visual feature variations
+
+Feature Lifecycle Management
+
+- Follow proper staging processes from development to general availability
+- Document feature flag purpose and intended lifecycle
+- Plan for feature flag retirement and cleanup
+- Implement proper versioning for feature flag definitions
+- Coordinate feature releases across frontend and backend systems
+- Set up proper monitoring for feature adoption and health
+
+Analytics and Feedback
+
+- Track feature usage properly with Bucket analytics
+- Implement feedback collection at appropriate interaction points
+- Design effective user feedback mechanisms for features
+- Analyze feature performance through metrics and KPIs
+- Use STARS framework for effective feature analysis
+- Connect feature metrics with business outcomes
+
+Security and Compliance
+
+- Implement proper access controls for feature management
+- Ensure sensitive configuration data is properly secured
+- Follow data privacy guidelines when tracking feature usage
+- Implement proper user identification and authentication
+- Handle entitlements and permissions correctly
+- Follow security best practices for client-side feature flags
+
+Testing and QA
+
+- Test feature flags in multiple environments
+- Implement automated tests for feature flag logic
+- Create test fixtures for different feature flag states
+- Simulate various targeting scenarios during testing
+- Ensure proper fallbacks for feature flag service failures
+- Test performance implications of feature flag implementation
+
+Integration with Development Workflow
+
+- Integrate feature flags with CI/CD pipelines
+- Implement proper feature flag documentation
+- Use feature flags for trunk-based development
+- Establish protocols for feature flag reviews
+- Coordinate feature flag management across teams
+- Implement proper change management for flag updates
+
+Follow Official Documentation
```

Comment:
Should we add actual links here to some of our docs related to SDK?

Reply:
They exist at the end.